### PR TITLE
cmd/openshift-install: Add 'destroy bootstrap' command

### PR DESCRIPTION
**Pull request by wking**
Using Terraform to remove all resources created by the bootstrap modules.  For this to work, all platforms must define a bootstrap module (and they all currently do).

I've also created a `LoadMetadata` helper to share the "retrieve the metadata from the asset directory" logic between the `destroy-cluster` and `destroy-bootstrap` logic.  The new helper lives in the cluster asset plackage close to the code that determines that file's location.

/assign @crawford

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
closes #200

File: pkg/destroy/bootstrap/bootstrap.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	"github.com/openshift/installer/data"
+	"github.com/openshift/installer/pkg/asset/cluster"
+	"github.com/openshift/installer/pkg/terraform"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// Destroy uses Terraform to remove bootstrap resources.
+func Destroy(dir string) (err error) {
+	tempDir, err := ioutil.TempDir("", "openshift-install-")
```
[the remark this answers is not in the document]
Comment (wking) [reply pkg/destroy/bootstrap/bootstrap.go]: > I think the Cluster asset also does this setup of moving the tfvars, tf files to temp and call terraform from that, do you thin kwe can create a setup func for both ?

I thought about that, but couldn't decide where to put it.  Should it just live in `data/unpack.go`?

[the remark this answers is not in the document]
Comment (wking) [reply pkg/destroy/bootstrap/bootstrap.go]: > `pkg/terraform/terraform.go` seems appropriate

I've pushed 640547a -> 69078c7, pushing the Terraform module unpacking and `terraform init` call down into a helper used by the `Apply` and `Destroy` functions.

**Comment by wking** [comment]
Rebased around #497 with 69078c7 -> 3f4fe57.

**Comment by wking** [comment]
We may want to do something to make this more gentle.  Maybe run some soft-shutdown script before firing the event.  As it stands, running `openshift-install delete bootstrap` after the `bootstrap-complete` event comes through leaves you with the cluster wondering where the bootstrap node went:

```console
$ KUBECONFIG=wking/auth/kubeconfig oc get nodes
NAME              STATUS     ROLES       AGE       VERSION
wking-bootstrap   NotReady   bootstrap   46m       v1.11.0+d4cacc0
wking-master-0    Ready      master      46m       v1.11.0+d4cacc0
$ KUBECONFIG=wking/auth/kubeconfig oc get node -o json wking-bootstrap | jq '.status.conditions[0]'
{
  "lastHeartbeatTime": "2018-10-21T04:37:20Z",
  "lastTransitionTime": "2018-10-21T04:42:24Z",
  "message": "Kubelet stopped posting node status.",
  "reason": "NodeStatusUnknown",
  "status": "Unknown",
  "type": "OutOfDisk"
}
```

What we want is to remove that node entirely.  You can currently do that by hand:

```console
$ KUBECONFIG=wking/auth/kubeconfig oc delete node wking-bootstrap
node "wking-bootstrap" deleted
```

but I'm not sure what the best way is to wire that into `delete bootstrap`.

**Pull request merged**